app/models/nic/managed_extensions.rb

```diff
@@ -0,0 +1,27 @@
+module Nic::ManagedExtensions
+  extend ActiveSupport::Concern
+
+  included do
+    after_validation :discovery_queue_rebuild_dns
+  end
+
+  def discovery_queue_rebuild_dns
+    return unless (dns? || dns6? || reverse_dns? || reverse_dns6?) && errors.empty? && Setting[:discovery_always_rebuild_dns]
+    return if self.host.new_record? # Discovered Hosts already exist, and new_records will break `find`
+    return unless self.host.type_changed? and ::Host::Base.find(self.host.id).type == "Host::Discovered"
+    return if self.pending_dns_record_changes?
+    logger.debug "Queuing DNS rebuild for #{self}"
+    queue.create(:name => _("Rebuild DNS for %s") % self, :priority => 10,
+                 :action => [self, :set_discovery_rebuild_dns])
```

Comment:
> The rebuild is only fired when there are no DNS changes pending, so it cannot collide. Therefore, I think 10 is the most appropriate priority, since this is the priority DNS changes are executed. 

Yeah, for some reason I misunderstood the whole motivation. I agree.

-- 
Later,
 Lukas #lzap Zapletal